src/components/Onboarding/OnboardingTos.tsx

```diff
@@ -81,12 +91,22 @@ export function OnboardingTos() {
             size="lg"
             onClick={handleStepComplete}
             loading={isLoading}
-            disabled={isLoadingRecaptcha || !recaptchaToken}
+            disabled={captchaState.status !== 'success'}
           >
             Accept
           </Button>
         </Group>
       )}
+      {env.NEXT_PUBLIC_CLOUDFLARE_TURNSTILE_SITEKEY && (
+        <Turnstile
+          options={{ size: 'invisible' }}
+          onSuccess={(token) => setCaptchaState({ status: 'success', token })}
+          onError={() => setCaptchaState({ status: 'error', token: null })}
+          onExpire={(token) => setCaptchaState({ status: 'expired', token })}
+          siteKey={env.NEXT_PUBLIC_CLOUDFLARE_TURNSTILE_SITEKEY}
+          injectScript={false}
+        />
+      )}
```

Comment:
I suggest we make a separate component that encompasses this on its own. Sounds more re-usable & no need to dirty other components with env checks which are ugly enough for what they are